### ISSUE TITLE
Incorrect calculation (returns negative number) of free disk space available on very large volume (AWS EFS) 

### ISSUE THREAD

**Issue by ComeMaes** [orientechnologies/orientdb#6688]
Hello, I am trying to use orientdb with Amazon EFS and are running into problems that seem related to the calculation of free disk space available.

## OrientDB Version, operating system, or hardware.
version 2.2.8
 Amazon Elastic File System volume

## Operating System
Linux (Debian on EC2 instances)

## Expected behavior and actual behavior
On a "virtually infinite" storage space such as AWS EFS (running `df` shows space available the orders of 8 exabytes), I expect Orientdb to take into account that very large amount of remaining space correctly. However, it returns a negative number as "free disk space available".
 
## Steps to reproduce the problem
The EFS volume is mounted in orientdb containers (official 2.2.8 image) to directory `orientdb/databases` 

If I mount the EFS volume on an instance and run `df -hT /home/ec2-user/mount_point/` I get the following:

`file system                                                                    Type  Size  Used  Avail. Used%  Mount`
`XXXX.efs.YYY.amazonaws.com:/   nfs4   8,0E     0     8,0E   0%       /home/ec2-user/mount_point`

When I run one server container, and run the console in another one to try to create a database, here's my console output:

`orientdb {server=remote:orientdb/}> create database remote:orientdb/test root root local`
`Creating database [remote:orientdb/test] using the storage type [plocal]...`
`Error: com.orientechnologies.orient.core.exception.OStorageException: Cannot create the remote storage: test`

`Error: com.orientechnologies.orient.core.exception.ODatabaseException: Cannot create database 'test'`
`	DB name="test"`
`	DB name="test"`

`Error: com.orientechnologies.orient.core.exception.OLowDiskSpaceException: Error occurred while executing a write operation to database 'test' due to limited free space on the disk (-8796093022208 MB). The database is now working in read-only mode. Please close the database (or stop OrientDB), make room on your hard drive and then reopen the database. The minimal required space is 256 MB. Required space is now set to 256MB (you can change it by setting parameter storage.diskCache.diskFreeSpaceLimit) .`
`	DB name="test"`

So there is that obvious negative free space left on the disk number. As EFS volumes are supposed to be virtually infinite, I suspect the 8,0E size given by the `df` command correspond to some kind of largest possible integer value. Therefore my initial supposition would be that orientdb misinterprets this number as signed, but I have not investigated this much deeper.

Any hints about this?

**Comment by laa** [comment]
@ComeMaes could you build latest version of 2.2.x branch (just execute command ```mvn clean package```) and create database with default settings.

**Comment by ComeMaes** [comment]
Hi @laa,

I've tried building from 2.2.x branch in an ec2 instance to be able to mount the EFS volume afterwards. 
`mvn clean package` failed on OrientDB Core due to failed tests so I had to skip them.

Once build successful, from `distribution/target` directory I ran:
`$ df -hT databases`
`FileSystem                                     Type Size Used Avail. Use%  Mounted on`
`XXXX.YYYYY.efs.ZZZZ.amazonaws.com:/ nfs4   8,0E       0  8,0E   0% /home/ec2-user/orient/orientdb/distribution/target/databases`

`$ ./orientdb-community-2.2.11-SNAPSHOT.dir/orientdb-community-2.2.11-SNAPSHOT/bin/console.sh`

`OrientDB console v.2.2.11-SNAPSHOT (build 2.2.x@r714c8e370eb986d81a451bdca0ca86af2e927a72; 2016-09-16 10:30:36+0000) www.orientdb.com
Type 'help' to display all the supported commands.
Installing extensions for GREMLIN language v.2.6.0`

`orientdb> create database remote:localhost/test root root plocal`

`Creating database [remote:localhost/test] using the storage type [plocal]...`
`Connecting to database [remote:localhost/test] with user 'root'...OK`
`Database created successfully.`

`Current database is: remote:localhost/test`
`orientdb {db=test}> `

So, the way it looks, the problem is solved. Will this fix be integrated in an upcoming release soon? Possibly a dockerized one, for my interests ;)
